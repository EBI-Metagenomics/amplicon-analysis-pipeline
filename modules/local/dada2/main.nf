
process DADA2 {
    // Run DADA2 pipeline including read-tracking
    tag "$meta.id"
    label "medium"
    // TODO: this env file doesn't exist
    conda "${moduleDir}/dada2_environment.yml"

    input:
    tuple val(meta), path(reads)

    output:
    tuple val(meta), path("*map.txt"), path("*asvs.fasta"), path("*_filt.fastq.gz"), emit: dada2_out
    path("*asv_counts.tsv"),                                                       emit: dada2_asv_counts
    tuple path("*chimeric.txt"), path("*matched.txt"),                             emit: dada2_stats
    
    script:
    if ( meta.single_end )
        """
        dada2.R ${meta.id} $reads
        """
    } else {
        """
        dada2.R ${meta.id} ${reads[0]} ${reads[1]}
        """
}
