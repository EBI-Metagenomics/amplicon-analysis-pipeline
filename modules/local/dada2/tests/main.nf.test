nextflow_process {

    name "Test Process DADA2"
    script "modules/local/dada2/main.nf"
    process "DADA2"
    tag "modules"
    tag "dada2"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    "16S-V3-V4",
                    [ 
                        file('${projectDir}/modules/local/dada2/tests/data/test_noambig_1.fastq.gz', checkIfExists: true),
                        file('${projectDir}/modules/local/dada2/tests/data/test_noambig_2.fastq.gz', checkIfExists: true) 
                    ]
                ]
                input[1] = file('${projectDir}/modules/local/dada2/tests/data/silva_trimmed.fa.gz', checkIfExists: true)
                input[2] = "DADA2-SILVA"
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(process.out.dada2_out.get(0).get(5)).readLines().size() == 20).match("taxa_tsv_length_eq_20") },
                { assert snapshot(process.out.dada2_out.get(0).get(2)).match() }
            )
        }

    }

}
