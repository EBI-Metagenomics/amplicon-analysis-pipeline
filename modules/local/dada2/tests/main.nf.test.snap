{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    [
                        "test_1_map.txt:md5,93a753df1317f72cb181045bc2be3949",
                        "test_2_map.txt:md5,93a753df1317f72cb181045bc2be3949"
                    ],
                    "test_asvs.fasta:md5,3150f757e54a9f7fcc911d44f6495b4e",
                    [
                        "test_1_filt.fastq.gz:md5,35819d59fab6649520fa36e63a405da4",
                        "test_2_filt.fastq.gz:md5,54d86d90378ecdb21e57e545f081f4a4"
                    ]
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-06T10:13:43.129627"
    }
}